### PR TITLE
prevents xenos catching thrown items they can't hold

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
prevents xenos catching thrown items they can't hold

## Why It's Good For The Game
xenos shouldn't be able to do this, same thing with the give stuff which was patched already

## Testing
threw some stuff at a xeno

## Changelog
:cl:
fix: xenos can no longer catch items they shouldn't be able to hold
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -8,9 +8,12 @@
 			if(istype(AM, /obj/item/twohanded))
 				if(get_inactive_hand())
 					return FALSE
+			throw_mode_off()
+			if(isalien(src))
+				AM.attack_alien(src)
+				return FALSE
```
[the remark this answers is not in the document]
Comment (Fox-McCloud) [reply code/modules/mob/living/carbon/carbon_defense.dm:14]: This is gross.

it should be:

```
/mob/living/carbon/alien/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum)
	..(AM, skipcatch = TRUE, hitpush = FALSE)
```

[the remark this answers is not in the document]
Comment (GDNgit) [reply code/modules/mob/living/carbon/carbon_defense.dm:14]: That'll stop em from catching everything

[the remark this answers is not in the document]
Comment (GDNgit) [reply code/modules/mob/living/carbon/carbon_defense.dm]: Applied

File: code/modules/mob/living/carbon/alien/alien_defense.dm [hunk 2]
```diff
@@ -1,5 +1,8 @@
 /mob/living/carbon/alien/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum)
-	..(AM, hitpush = FALSE)
+	if(!skipcatch && in_throw_mode && !HAS_TRAIT(src, TRAIT_HANDS_BLOCKED) && !restrained())
+		throw_mode_off()
+		AM.attack_alien(src)
```
Comment (lewcc) [inline code/modules/mob/living/carbon/alien/alien_defense.dm:4]: I still feel a little sus about this. How many items can aliens actually pick up or catch in general? Do they *need* to be able to catch anything?

Comment (GDNgit) [reply code/modules/mob/living/carbon/alien/alien_defense.dm:4]: Catching facehuggers does actually come up, and more stuff will probably be added in the future, especially when space xenoing
Also if you like throw someone at a xeno... what do you expect

Comment (lewcc) [reply code/modules/mob/living/carbon/alien/alien_defense.dm:4]: At that point it feels like we should just add a trait to things that xenos can pick up. Calling the attack_alien on anything thrown still feels a little off to me, but I'm curious what other folks think

Comment (S34NW) [reply code/modules/mob/living/carbon/alien/alien_defense.dm:4]: Guh, I'm torn. Happy for a trait-based approach since there are so few examples here.

Comment (GDNgit) [reply code/modules/mob/living/carbon/alien/alien_defense.dm]: moved to a trait solution

**Review by S34NW — APPROVED** [review]

**Pull request merged**